src/jsMain/kotlin/baaahs/sim/ui/NetworkPanel.kt

```diff
@@ -0,0 +1,96 @@
+package baaahs.sim.ui
+
+import baaahs.Observable
+import baaahs.SheepSimulator
+import baaahs.sim.FakeNetwork
+import kotlinx.css.*
+import kotlinx.html.id
+import kotlinx.html.js.onClickFunction
+import react.RBuilder
+import react.RComponent
+import react.RProps
+import react.RState
+import react.dom.tbody
+import react.dom.td
+import react.dom.tr
+import styled.css
+import styled.styledTable
+import styled.styledTd
+import styled.styledTh
+
+class NetworkPanel(props: Props) : RComponent<NetworkPanel.Props, NetworkPanel.State>(props), Observable.Observer {
+    private var changedSinceRender: Boolean = false
+
+    override fun componentDidMount() {
+        props.simulator?.network?.addObserver(this)
+    }
+
+    override fun componentWillUnmount() {
+        props.simulator?.network?.removeObserver(this)
+    }
+
+    override fun componentWillUpdate(nextProps: Props, nextState: State) {
+        props.simulator?.network?.removeObserver(this)
+        nextProps.simulator?.network?.addObserver(this)
+    }
+
+    override fun notifyChanged() {
+        if (!changedSinceRender) {
+            changedSinceRender = true
+            setState({ s -> s }) {
```

Comment:
`setState()` here always causes a _synchronous_ call to render, so _every single packet_ causes a full render, which is super slow.